integration-tests/src/tests/client/block_corruption.rs

```diff
@@ -99,3 +101,137 @@ fn change_shard_id_to_invalid() {
+
+    // Technically we don't have to do first step, because of Ok("End") check.
+    // But we may change that part in the future for some reason, and I want this part to be reliable.
+
+    // get BIT index modulo BIT length of block
+    let mod_idx = corrupted_bit_idx % (block_vec.len() * 8);
+    // get index of BYTE that we are changing
+    let byte_idx = mod_idx / 8;
+    // get index of BIT in that BYTE that we are flipping
+    let bit_idx = mod_idx % 8;
+    // flip that BIT
+    block_vec[byte_idx] ^= 1 << bit_idx;
+
+    if let Ok(mut corrupt_block) = Block::try_from_slice(block_vec.as_slice()) {
+        corrupt_block.mut_header().get_mut().inner_rest.block_body_hash =
+            corrupt_block.compute_block_body_hash().unwrap();
+        corrupt_block.mut_header().resign(&InMemoryValidatorSigner::from_seed(
+            "test0".parse().unwrap(),
+            KeyType::ED25519,
+            "test0",
+        ));
+
+        if is_breaking_block_change(&correct_block, &corrupt_block) {
+            match env.clients[0].process_block_test(corrupt_block.clone().into(), Provenance::NONE)
+            {
+                Ok(_) => {
+                    return Err(anyhow::anyhow!(
+                        "Was able to process default block with {} bit switched.",
+                        corrupted_bit_idx
+                    ));
+                }
+                Err(e) => {
+                    if let Err(e) =
+                        env.clients[0].process_block_test(correct_block.into(), Provenance::NONE)
+                    {
+                        return Err(anyhow::anyhow!("Was unable to process default block after attempting to process default block with {} bit switched. {}",
+                        corrupted_bit_idx, e)
+                        );
+                    }
+                    return Ok(e.into());
+                }
+            }
+        } else {
+            return Ok(anyhow::anyhow!("Not a breaking change"));
+        }
+    }
+    return Ok(anyhow::anyhow!("Corrupt block didn't parse"));
+}
+
+#[test]
+#[cfg_attr(not(feature = "expensive_tests"), ignore)]
+fn check_process_flipped_block_fails() {
+    init_test_logger();
+    let mut corrupted_bit_idx = 0;
+    let mut errs = vec![];
+    let mut oks = vec![];
+    loop {
+        let res = check_process_flipped_block_fails_on_bit(corrupted_bit_idx);
+        if let Ok(res) = &res {
+            if res.to_string() == "End" {
+                break;
+            }
+        }
+        match res {
+            Err(e) => errs.push(e),
+            Ok(o) => oks.push(o),
+        }
+        corrupted_bit_idx += 1;
+    }
+    tracing::info!("All of the Errors:");
+    for err in &errs {
+        tracing::info!("{:?}", err);
+    }
+    tracing::info!("{}", ["-"; 100].concat());
+    tracing::info!("All of the Oks:");
+    for ok in oks {
+        tracing::info!("{:?}", ok);
+    }
+    assert!(errs.is_empty());
```

Comment:
This test checks that at least one bit flip out of the whole block fails?

Reply:
`errs` and `oks` are confusing names. If you check `check_process_flipped_block_fails_on_bit`, it returns `Result<anyhow::Error, anyhow::Error>`. If processing of corrupt block fails (as expected) it returns `Ok(processing_error)` and `processing_error` is written to `oks`. If processing of corrupt block succeeds it returns `Err(explanation_string)` and `explanation_string` is written to `errs`.
So here we expect that `check_process_flipped_block_fails_on_bit` never returned an `Err`, so processing of every corrupt block failed.
I will add more comments to make it less confusing.

Reply:
Oh! Not it is clear.
In this case, I think that we can add another check to have expected errors other than `Ok(anyhow::anyhow!("Not a breaking change"))`. It would be weird not to have any